netlogo-core/src/main/api/LabProtocol.scala

```diff
@@ -14,7 +14,7 @@ case class LabProtocol(name: String,
                     metrics: List[String],
                     valueSets: List[RefValueSet])
 {
-  def countRuns = repetitions * valueSets.map(_.toList.size).product
+  def countRuns: Int = repetitions * valueSets.map(_.toList.size: Int).foldLeft(1)((accum, n) => accum * n)
```

Comment:
I don't understand the intent behind this change. Why is this preferable to the old code?

Reply:
The change is, mostly, for style, preferring foldLeft over product. There is a small advantage in this context, such as providing space to add error handling for integer overflow for each multiplication. But it is not necessary for this fix.

Reply:
what's wrong, style-wise, with `product`?

Reply:
The issue I had with product is that .product can apply to any type T with an implicit Numeric[T] implementation. So it would apply for Long, Float, and BigInt, which could "change" the type/behavior. With foldLeft, there are two type constraints to satisfy. The accumulator and the result must have the same type.
```
 foldLeft[B](z: B)(op: (B, A) ⇒ B): B 
```
This would, also, allow adding Options for error handling.

Reply:
@charlesres, the `foldLeft` isn't giving you any added protection against type coercion.  The type ascription (of `Int`) on `def countRuns` already guarantees that you'll get a type error if `product` is run on non-`Int`s.

Regarding the `Option` stuff, do we have any plans of expanding this to more-general monoids?  If we do that, can't we handle that when it comes up?  I don't see any problem with the old use of `product`.  It seems a lot more stylistically pleasant to me than manually writing out the fold.

Reply:
@TheBizzle, I agree that product would be better suited in that case. As for plans, I don't think there are plans to expand that case, so I'll keep product in LabProtocol.scala instead of foldLeft.